sdk/go/common/resource/plugin/analyzer_server.go

```diff
@@ -0,0 +1,429 @@
+			KeepUnknowns:     true,
+			KeepSecrets:      true,
+			SkipInternalKeys: false,
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		return &pulumirpc.Remediation{
+			PolicyName:        r.PolicyName,
+			PolicyPackName:    r.PolicyPackName,
+			PolicyPackVersion: r.PolicyPackVersion,
+			Description:       r.Description,
+			Properties:        mprops,
+			Diagnostic:        r.Diagnostic,
+		}, nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &pulumirpc.RemediateResponse{
+		Remediations:  remediations,
+		NotApplicable: marshalPolicyNotApplicables(res.NotApplicable),
+	}, nil
+}
+
+func (a *analyzerServer) GetAnalyzerInfo(context.Context, *emptypb.Empty) (*pulumirpc.AnalyzerInfo, error) {
+	info, err := a.analyzer.GetAnalyzerInfo()
+	if err != nil {
+		return nil, err
+	}
+
+	policies := slice.Map(info.Policies, func(p AnalyzerPolicyInfo) *pulumirpc.PolicyInfo {
+		return &pulumirpc.PolicyInfo{
+			Name:             p.Name,
+			DisplayName:      p.DisplayName,
+			Description:      p.Description,
+			Message:          p.Message,
+			EnforcementLevel: marshalEnforcementLevel(p.EnforcementLevel),
+			ConfigSchema:     marshalConfigSchema(p.ConfigSchema),
+			PolicyType:       marshalPolicyType(p.Type),
+			Severity:         marshalPolicySeverity(p.Severity),
+			Framework:        marshalComplianceFramework(p.Framework),
+			Tags:             p.Tags,
+			RemediationSteps: p.RemediationSteps,
+			Url:              p.URL,
+		}
+	})
+
+	initialConfig := make(map[string]*pulumirpc.PolicyConfig)
+	for k, v := range info.InitialConfig {
+		properties, err := structpb.NewStruct(v.Properties)
+		contract.AssertNoErrorf(err, "marshaling initial config properties for policy %s", k)
+		initialConfig[k] = &pulumirpc.PolicyConfig{
+			EnforcementLevel: marshalEnforcementLevel(v.EnforcementLevel),
+			Properties:       properties,
+		}
+	}
+
+	return &pulumirpc.AnalyzerInfo{
+		Name:           info.Name,
+		DisplayName:    info.DisplayName,
+		Version:        info.Version,
+		SupportsConfig: info.SupportsConfig,
+		Policies:       policies,
+		InitialConfig:  initialConfig,
+		Description:    info.Description,
+		Readme:         info.Readme,
+		Provider:       info.Provider,
+		Tags:           info.Tags,
+		Repository:     info.Repository,
+	}, nil
+}
+
+func (a *analyzerServer) GetPluginInfo(context.Context, *emptypb.Empty) (*pulumirpc.PluginInfo, error) {
+	info, err := a.analyzer.GetPluginInfo()
+	if err != nil {
+		return nil, err
+	}
+	return &pulumirpc.PluginInfo{Version: info.Version.String()}, nil
+}
+
+func (a *analyzerServer) Configure(_ context.Context, req *pulumirpc.ConfigureAnalyzerRequest) (*emptypb.Empty, error) {
+	config, err := convertPolicyConfig(req.GetPolicyConfig())
+	if err != nil {
+		return nil, err
+	}
+
+	if err := a.analyzer.Configure(config); err != nil {
+		return nil, err
+	}
+
+	return &emptypb.Empty{}, nil
+}
+
+func (a *analyzerServer) Handshake(
```

Comment:
Should we add these to the Analyzer interface. We did for Provider.

Reply:
Maybe (probably). I'll leave that for a possible follow-up.